otter/models/cass.py

```diff
@@ -1231,15 +1232,18 @@ def _delete_group():
             return d
 
         def _delete_lock_znode(result):
-            d = self.kz_client.delete(LOCK_PATH + '/' + self.uuid,
-                                      recursive=True)
```

Comment:
There _is_ stuff inside of these lock directories, and the problem is that deleting them recursively while someone else is trying to acquire the lock (or while it's held) can wreak havoc. There is content inside the lock directory only while the lock is being acquired or held.

Deleting it with recursive=False means that, if the lock is in use, the deletion will fail, which is why I added the 5 minutes of retries every 5 seconds -- this is very likely to delete the lock given our usage patterns, but will eventually give up and simply leave it there.

This PR is basically a stopgap to introducing a more persistent lock-cleanup solution based on the idea of a "child reaper" a la Curator (mentioned in the issue associated with this PR). These "child reapers" act very similar to this code, but they do it for all entries in a directory -- i.e., they list all children and constantly try to delete them non-recursively. This is pretty much the only way to definitely keep a directory full of locks clean, according to my discussion with ZK community members.